src/confluent_kafka/aio/producer/_message_batch.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2025 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+from typing import List, Any, Optional
+
+
+# Create immutable MessageBatch value object
+MessageBatch = namedtuple('MessageBatch', [
+    'topic',        # str: Target topic for this batch
+    'messages',     # tuple: Prepared message dictionaries with callbacks
+    'futures',      # tuple: asyncio.Future objects to resolve
+    'callbacks'     # tuple: User callback functions (can be None)
+])
+
+
+def create_message_batch(topic: str, messages: List[dict], futures: List[Any], callbacks: List[Optional[callable]]) -> MessageBatch:
```

Comment:
Can set the types to `Sequence` instead of `List` to allow for tuples

Reply:
Not entirely sure this method is needed if it's just call the class constructor, which could do the argument cleanup / copies in an __init__ if it was a NamedTuple Class